### ISSUE TITLE
`function_missing` and `pub_module_level_const_missing` fire when removing items, but re-exporting the same items from a future version (semver-trick)

### ISSUE THREAD

**Issue by JSorngard** [obi1kenobi/cargo-semver-checks#1586]
### Which lint or lints are the issue

I have a crate where I recently published version 2.0.0. In order for version 1.x.y to still get future bug fixes and improvements to functions I thought to use the [semver-trick](https://github.com/dtolnay/semver-trick) to re-export functions and constants from version 2 in version 1.

As a result version 1.x.y+1 removes functions and constants from version 1.x.y, but re-exports those same functions and constants from version 2.0.0. I do not believe this should be a problem, but `cargo-semver-checks` triggers the `pub_module_level_const_missing` and `function_missing` lints.

Here's a simplified description:

published version 1.x.y contains:  
- function `pub fn foo() -> i32`,
- constant `pub const BAR`,

published version 2.0.0 contains:  
- function `pub fn foo() -> i32`,
- constant `pub const BAR`,

semver-tricky version 1.x.y+1 should contain:  
- `pub use version2::foo`,
- `pub use version2::BAR`.

The actual names of the constants and functions are listed here to ease comparison if you decide to look into this:

- constants `OMEGA` and `NEG_INV_E` exist in 1.x.y, but 1.x.y+1 re-exports the ones from 2.0.0 which triggers `pub_module_level_const_missing`.
- functions `lambert_wm1`, `sp_lambert_wm1`, `lambert_w0f`, `lambert_w0`, `sp_lambert_w0` and `lambert_wm1f` exist in 1.x.y, but 1.x.y+1 re-exports the ones from 2.0.0 which triggers `function_missing`.

### Known issues that might be causing this

- [x] Is the flagged item defined in another crate, or a re-export of such an item? ([#638](https://github.com/obi1kenobi/cargo-semver-checks/issues/638))

### Steps to reproduce the bug with the above code

I don't know a good way to produce a minimal reproduction given the need to have packages on crates.io, so I link to relevant versions that illustrate the problem. The crate is quite small, so this should hopefully be an understandable case.

- Latest published version 1: https://crates.io/crates/lambert_w/1.2.34,  
- Published version 2: https://crates.io/crates/lambert_w/2.0.0,  
- The version 1 on github that I wish to publish where I remove several functions and constants and re-export them from version 2: https://github.com/JSorngard/lambert_w/tree/semver-checks-issue.

Basically, all files in the crate are replaced by these re-exports: https://github.com/JSorngard/lambert_w/blob/014b57274891b3a0a238c9e723bd7b18a8c90f16/src/lib.rs/#L32

If you checkout the version of the crate at https://github.com/JSorngard/lambert_w/tree/semver-checks-issue and run `cargo-semver-checks` on it the lints should trigger.

### Actual Behaviour

- `pub_module_level_const_missing` triggers and says that `BAR` is gone.
- `function_missing` triggers and says that `foo()` is gone.

### Expected Behaviour

No lint should trigger, as items with identical names, APIs and types are re-exported from a future version of the crate.

### Verbose Lint Output

    Updating crates.io index
     Locking 6 packages to latest compatible versions
    Building lambert_w v1.2.35 (current)
 Documenting lambert_w v1.2.35 (D:\Repositories\Rust\lambert_w)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.57s
       Built [   0.807s] (current)
     Parsing lambert_w v1.2.35 (current)
      Parsed [   0.002s] (current)
     Parsing lambert_w v1.2.34 (baseline, cached)
      Parsed [   0.002s] (baseline)
    Checking lambert_w v1.2.34 -> v1.2.35 (patch change)
    Starting 221 checks, 24 unnecessary on 16 threads
        PASS [   0.001s]       major        attribute_proc_macro_missing
        PASS [   0.002s]       major        auto_trait_impl_removed
        PASS [   0.002s]       major        constructible_struct_adds_field
        PASS [   0.001s]       major        constructible_struct_adds_private_field
        PASS [   0.000s]       major        constructible_struct_changed_type
        PASS [   0.000s]       major        copy_impl_added
        PASS [   0.001s]       major        declarative_macro_missing
        PASS [   0.000s]       major        derive_helper_attr_removed
        PASS [   0.000s]       major        derive_proc_macro_missing
        PASS [   0.001s]       major        derive_trait_impl_removed
        PASS [   0.000s]       major        enum_changed_kind
        PASS [   0.001s]       major        enum_discriminants_undefined_non_exhaustive_variant
        PASS [   0.001s]       major        enum_discriminants_undefined_non_unit_variant
        PASS [   0.001s]       major        enum_marked_non_exhaustive
        PASS [   0.000s]       major        enum_missing
        PASS [   0.000s]       minor        enum_must_use_added
        PASS [   0.001s]       major        enum_no_repr_variant_discriminant_changed
        PASS [   0.001s]       major        enum_non_exhaustive_tuple_variant_changed_kind
        PASS [   0.000s]       major        enum_now_doc_hidden
        PASS [   0.002s]       major        enum_repr_int_changed
        PASS [   0.001s]       major        enum_repr_int_removed
        PASS [   0.000s]       major        enum_repr_transparent_removed
        PASS [   0.001s]       major        enum_repr_variant_discriminant_changed
        PASS [   0.001s]       major        enum_struct_variant_changed_kind
        PASS [   0.000s]       major        enum_struct_variant_field_added
        PASS [   0.001s]       minor        enum_struct_variant_field_marked_deprecated
        PASS [   0.000s]       major        enum_struct_variant_field_missing
        PASS [   0.002s]       major        enum_struct_variant_field_now_doc_hidden
        PASS [   0.000s]       major        enum_tuple_variant_changed_kind
        PASS [   0.000s]       major        enum_tuple_variant_field_added
        PASS [   0.000s]       minor        enum_tuple_variant_field_marked_deprecated
        PASS [   0.000s]       major        enum_tuple_variant_field_missing
        PASS [   0.001s]       major        enum_tuple_variant_field_now_doc_hidden
        PASS [   0.000s]       major        enum_unit_variant_changed_kind
        PASS [   0.001s]       major        enum_variant_added
        PASS [   0.001s]       minor        enum_variant_marked_deprecated
        PASS [   0.000s]       major        enum_variant_marked_non_exhaustive
        PASS [   0.000s]       major        enum_variant_missing
        PASS [   0.000s]       major        exported_function_abi_no_longer_unwind
        PASS [   0.000s]       major        exported_function_abi_now_unwind
        PASS [   0.000s]       major        exported_function_changed_abi
        PASS [   0.000s]       major        exported_function_now_returns_unit
        PASS [   0.000s]       major        exported_function_parameter_count_changed
        PASS [   0.001s]       major        exported_function_requires_more_target_features
        PASS [   0.000s]       major        exported_function_return_value_added
        PASS [   0.001s]       major        exported_function_target_feature_added
        PASS [   0.000s]       major        feature_missing
        PASS [   0.001s]       major        feature_no_longer_enables_feature
        PASS [   0.001s]       major        feature_not_enabled_by_default
        PASS [   0.000s]       major        function_abi_no_longer_unwind
        PASS [   0.001s]       major        function_abi_now_unwind
        PASS [   0.000s]       major        function_changed_abi
        PASS [   0.001s]       major        function_const_generic_reordered
        PASS [   0.000s]       major        function_const_removed
        PASS [   0.000s]       major        function_export_name_changed
        PASS [   0.002s]       major        function_generic_type_reordered
        PASS [   0.000s]       major        function_like_proc_macro_missing
        PASS [   0.000s]       minor        function_marked_deprecated
        FAIL [   0.000s]       major        function_missing
        PASS [   0.000s]       minor        function_must_use_added
        PASS [   0.000s]       major        function_now_doc_hidden
        PASS [   0.001s]       major        function_now_returns_unit
        PASS [   0.001s]       major        function_parameter_count_changed
        PASS [   0.001s]       major        function_requires_different_const_generic_params
        PASS [   0.001s]       major        function_requires_different_generic_type_params
        PASS [   0.000s]       major        function_unsafe_added
        PASS [   0.000s]       minor        global_value_marked_deprecated
        PASS [   0.000s]       major        inherent_associated_const_now_doc_hidden
        PASS [   0.001s]       major        inherent_associated_pub_const_missing
        PASS [   0.000s]       major        inherent_method_changed_abi
        PASS [   0.001s]       major        inherent_method_const_generic_reordered
        PASS [   0.001s]       major        inherent_method_const_removed
        PASS [   0.001s]       major        inherent_method_generic_type_reordered
        PASS [   0.000s]       major        inherent_method_missing
        PASS [   0.000s]       minor        inherent_method_must_use_added
        PASS [   0.000s]       major        inherent_method_no_longer_unwind
        PASS [   0.000s]       major        inherent_method_now_doc_hidden
        PASS [   0.000s]       major        inherent_method_now_returns_unit
        PASS [   0.000s]       major        inherent_method_now_unwind
        PASS [   0.001s]       major        inherent_method_unsafe_added
        PASS [   0.000s]       minor        macro_marked_deprecated
        PASS [   0.000s]       major        macro_no_longer_exported
        PASS [   0.001s]       major        macro_now_doc_hidden
        PASS [   0.000s]       major        method_export_name_changed
        PASS [   0.001s]       major        method_no_longer_has_receiver
        PASS [   0.001s]       major        method_parameter_count_changed
        PASS [   0.001s]       major        method_receiver_mut_ref_became_owned
        PASS [   0.001s]       major        method_receiver_ref_became_mut
        PASS [   0.001s]       major        method_receiver_ref_became_owned
        PASS [   0.001s]       major        method_receiver_type_changed
        PASS [   0.001s]       major        method_requires_different_const_generic_params
        PASS [   0.001s]       major        method_requires_different_generic_type_params
        PASS [   0.000s]       major        module_missing
        PASS [   0.000s]       major        non_exhaustive_struct_changed_type
        PASS [   0.001s]       major        partial_ord_enum_struct_variant_fields_reordered
        PASS [   0.001s]       major        partial_ord_enum_variants_reordered
        PASS [   0.000s]       major        partial_ord_struct_fields_reordered
        PASS [   0.000s]       minor        proc_macro_marked_deprecated
        PASS [   0.000s]       major        proc_macro_now_doc_hidden
        PASS [   0.000s]       minor        pub_api_sealed_trait_became_unconditionally_sealed
        PASS [   0.000s]       minor        pub_api_sealed_trait_became_unsealed
        PASS [   0.000s]       major        pub_api_sealed_trait_method_receiver_added
        PASS [   0.000s]       major        pub_api_sealed_trait_method_receiver_mut_ref_became_ref
        PASS [   0.000s]       major        pub_api_sealed_trait_method_return_value_added
        PASS [   0.000s]       major        pub_api_sealed_trait_method_target_feature_removed
        FAIL [   0.000s]       major        pub_module_level_const_missing
        PASS [   0.000s]       major        pub_module_level_const_now_doc_hidden
        PASS [   0.000s]       major        pub_static_missing
        PASS [   0.000s]       major        pub_static_mut_now_immutable
        PASS [   0.000s]       major        pub_static_now_doc_hidden
        PASS [   0.001s]       major        pub_static_now_mutable
        PASS [   0.002s]       major        repr_align_added
        PASS [   0.001s]       major        repr_align_changed
        PASS [   0.002s]       major        repr_align_removed
        PASS [   0.001s]       major        repr_c_enum_struct_variant_fields_reordered
        PASS [   0.001s]       major        repr_c_plain_struct_fields_reordered
        PASS [   0.001s]       major        repr_c_removed
        PASS [   0.000s]       major        repr_packed_added
        PASS [   0.001s]       major        repr_packed_changed
        PASS [   0.000s]       major        repr_packed_removed
        PASS [   0.000s]       major        safe_function_requires_more_target_features
        PASS [   0.000s]       major        safe_function_target_feature_added
        PASS [   0.001s]       major        safe_inherent_method_requires_more_target_features
        PASS [   0.002s]       major        safe_inherent_method_target_feature_added
        PASS [   0.000s]       major        sized_impl_removed
        PASS [   0.000s]       major        static_became_unsafe
        PASS [   0.000s]       minor        struct_field_marked_deprecated
        PASS [   0.000s]       major        struct_marked_non_exhaustive
        PASS [   0.000s]       major        struct_missing
        PASS [   0.000s]       minor        struct_must_use_added
        PASS [   0.000s]       major        struct_now_doc_hidden
        PASS [   0.000s]       major        struct_pub_field_missing
        PASS [   0.000s]       major        struct_pub_field_now_doc_hidden
        PASS [   0.001s]       major        struct_repr_transparent_removed
        PASS [   0.000s]       major        struct_with_no_pub_fields_changed_type
        PASS [   0.000s]       major        struct_with_pub_fields_changed_type
        PASS [   0.001s]       major        trait_added_supertrait
        PASS [   0.002s]       major        trait_allows_fewer_const_generic_params
        PASS [   0.000s]       major        trait_allows_fewer_generic_type_params
        PASS [   0.000s]       major        trait_associated_const_added
        PASS [   0.000s]       major        trait_associated_const_default_removed
        PASS [   0.000s]       minor        trait_associated_const_marked_deprecated
        PASS [   0.000s]       major        trait_associated_const_now_doc_hidden
        PASS [   0.001s]       major        trait_associated_type_added
        PASS [   0.001s]       major        trait_associated_type_default_removed
        PASS [   0.001s]       minor        trait_associated_type_marked_deprecated
        PASS [   0.001s]       major        trait_associated_type_now_doc_hidden
        PASS [   0.000s]       major        trait_changed_kind
        PASS [   0.001s]       major        trait_const_generic_reordered
        PASS [   0.001s]       major        trait_generic_type_reordered
        PASS [   0.000s]       minor        trait_marked_deprecated
        PASS [   0.001s]       major        trait_method_added
        PASS [   0.000s]       major        trait_method_changed_abi
        PASS [   0.001s]       major        trait_method_const_generic_reordered
        PASS [   0.000s]       major        trait_method_default_impl_removed
        PASS [   0.000s]       major        trait_method_generic_type_reordered
        PASS [   0.000s]       minor        trait_method_marked_deprecated
        PASS [   0.000s]       major        trait_method_missing
        PASS [   0.001s]       major        trait_method_no_longer_has_receiver
        PASS [   0.001s]       major        trait_method_no_longer_unwind
        PASS [   0.001s]       major        trait_method_now_doc_hidden
        PASS [   0.002s]       major        trait_method_now_returns_unit
        PASS [   0.001s]       major        trait_method_now_unwind
        PASS [   0.000s]       major        trait_method_parameter_count_changed
        PASS [   0.000s]       major        trait_method_receiver_added
        PASS [   0.002s]       major        trait_method_receiver_mut_ref_became_owned
        PASS [   0.000s]       major        trait_method_receiver_mut_ref_became_ref
        PASS [   0.000s]       major        trait_method_receiver_owned_became_mut_ref
        PASS [   0.000s]       major        trait_method_receiver_owned_became_ref
        PASS [   0.001s]       major        trait_method_receiver_ref_became_mut
        PASS [   0.001s]       major        trait_method_receiver_ref_became_owned
        PASS [   0.002s]       major        trait_method_receiver_type_changed
        PASS [   0.000s]       major        trait_method_requires_different_const_generic_params
        PASS [   0.000s]       major        trait_method_requires_different_generic_type_params
        PASS [   0.001s]       major        trait_method_return_value_added
        PASS [   0.000s]       major        trait_method_target_feature_removed
        PASS [   0.001s]       major        trait_method_unsafe_added
        PASS [   0.001s]       major        trait_method_unsafe_removed
        PASS [   0.001s]       major        trait_mismatched_generic_lifetimes
        PASS [   0.000s]       major        trait_missing
        PASS [   0.000s]       minor        trait_must_use_added
        PASS [   0.000s]       major        trait_newly_sealed
        PASS [   0.000s]       major        trait_no_longer_dyn_compatible
        PASS [   0.000s]       major        trait_now_doc_hidden
        PASS [   0.001s]       major        trait_removed_associated_constant
        PASS [   0.001s]       major        trait_removed_associated_type
        PASS [   0.000s]       major        trait_removed_supertrait
        PASS [   0.001s]       major        trait_requires_more_const_generic_params
        PASS [   0.000s]       major        trait_requires_more_generic_type_params
        PASS [   0.000s]       major        trait_unsafe_added
        PASS [   0.000s]       major        trait_unsafe_removed
        PASS [   0.000s]       major        tuple_struct_to_plain_struct
        PASS [   0.000s]       major        type_allows_fewer_const_generic_params
        PASS [   0.001s]       major        type_allows_fewer_generic_type_params
        PASS [   0.001s]       minor        type_associated_const_marked_deprecated
        PASS [   0.000s]       major        type_const_generic_reordered
        PASS [   0.001s]       major        type_generic_type_reordered
        PASS [   0.001s]       minor        type_marked_deprecated
        PASS [   0.001s]       minor        type_method_marked_deprecated
        PASS [   0.000s]       major        type_mismatched_generic_lifetimes
        PASS [   0.001s]       major        type_requires_more_const_generic_params
        PASS [   0.001s]       major        type_requires_more_generic_type_params
        PASS [   0.000s]       minor        unconditionally_sealed_trait_became_pub_api_sealed
        PASS [   0.000s]       minor        unconditionally_sealed_trait_became_unsealed
        PASS [   0.000s]       major        union_changed_kind
        PASS [   0.001s]       major        union_changed_to_incompatible_struct
        PASS [   0.001s]       major        union_field_added_with_all_pub_fields
        PASS [   0.001s]       major        union_field_added_with_non_pub_fields
        PASS [   0.001s]       major        union_field_missing
        PASS [   0.000s]       major        union_missing
        PASS [   0.000s]       minor        union_must_use_added
        PASS [   0.000s]       major        union_now_doc_hidden
        PASS [   0.001s]       major        union_pub_field_now_doc_hidden
        PASS [   0.000s]       major        union_with_multiple_pub_fields_changed_to_struct
        PASS [   0.000s]       major        unit_struct_changed_kind
        PASS [   0.000s]       major        unsafe_function_requires_more_target_features
        PASS [   0.001s]       major        unsafe_function_target_feature_added
        PASS [   0.001s]       major        unsafe_inherent_method_requires_more_target_features
        PASS [   0.000s]       major        unsafe_inherent_method_target_feature_added
        PASS [   0.001s]       major        unsafe_trait_method_requires_more_target_features
        PASS [   0.000s]       major        unsafe_trait_method_target_feature_added
     Checked [   0.009s] 221 checks: 219 pass, 2 fail, 0 warn, 24 skip

--- failure function_missing: pub fn removed or renamed ---

Description:
A publicly-visible function cannot be imported by its prior path. A `pub use` may have been removed, or the function itself may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.46.0/src/lints/function_missing.ron

Failed in:
  function lambert_w::lambert_wm1, previously in file C:\Users\USERNAME\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\lambert_w-1.2.34\src\dwm1c.rs:57
  function lambert_w::sp_lambert_wm1, previously in file C:\Users\USERNAME\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\lambert_w-1.2.34\src\swm1.rs:57
  function lambert_w::lambert_w0f, previously in file C:\Users\USERNAME\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\lambert_w-1.2.34\src\sw0f.rs:56
  function lambert_w::lambert_w0, previously in file C:\Users\USERNAME\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\lambert_w-1.2.34\src\dw0c.rs:58
  function lambert_w::sp_lambert_w0, previously in file C:\Users\USERNAME\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\lambert_w-1.2.34\src\sw0.rs:58
  function lambert_w::lambert_wm1f, previously in file C:\Users\USERNAME\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\lambert_w-1.2.34\src\swm1f.rs:56

--- failure pub_module_level_const_missing: pub module-level const is missing ---

Description:
A public const is missing or renamed
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.46.0/src/lints/pub_module_level_const_missing.ron

Failed in:
  NEG_INV_E in file C:\Users\USERNAME\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\lambert_w-1.2.34\src\lib.rs:61
  OMEGA in file C:\Users\USERNAME\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\lambert_w-1.2.34\src\lib.rs:89

     Summary semver requires new major version: 2 major and 0 minor checks failed
    Finished [   2.532s] lambert_w

### Generated System Information

System information:
-------------------
#### Software version

cargo-semver-checks 0.46.0 (097412d)

#### Operating system

- OS: Windows 11 Pro
- Kernel: 26200

#### Command-line

```bash
C:\Users\USERNAME\.cargo\bin\cargo-semver-checks.exe semver-checks --bugreport
```

#### cargo version

```
> cargo -V
cargo 1.93.0 (083ac5135 2025-12-15)
```

#### Compile time information

- Profile: release
- Target triple: x86_64-pc-windows-msvc
- Family: windows
- OS: windows
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: cmpxchg16b,fxsr,sse,sse2,sse3
- Host: x86_64-pc-windows-msvc

Cargo build configuration:
--------------------------
[alias]
br = ["build", "--release"]
brl = ["build", "--profile", "release-lto"]
rr = ["run", "--release"]
rrl = ["run", "--profile", "release-lto"]
tr = ["test", "--release"]
trl = ["test", "--profile", "release-lto"]

[build]
rustflags = ["-C", "target-cpu=native"]

Please file an issue on GitHub reporting your bug.
Consider adding the diagnostic information above, either manually or automatically through the link below:

https://github.com/obi1kenobi/cargo-semver-checks/issues/new?labels=C-bug&template=3-bug-report.yml&sys-info=%23%23%23%23%20Software%20version%0A%0Acargo-semver-checks%200.46.0%20%28097412d%29%0A%0A%23%23%23%23%20Operating%20system%0A%0A-%20OS%3A%20Windows%2011%20Pro%0A-%20Kernel%3A%2026200%0A%0A%23%23%23%23%20Command-line%0A%0A%60%60%60bash%0AC%3A%5CUsers%5CUSERNAME%5C.cargo%5Cbin%5Ccargo-semver-checks.exe%20semver-checks%20--bugreport%20%0A%60%60%60%0A%0A%23%23%23%23%20cargo%20version%0A%0A%60%60%60%0A%3E%20cargo%20-V%20%0Acargo%201.93.0%20%28083ac5135%202025-12-15%29%0A%60%60%60%0A%0A%23%23%23%23%20Compile%20time%20information%0A%0A-%20Profile%3A%20release%0A-%20Target%20triple%3A%20x86_64-pc-windows-msvc%0A-%20Family%3A%20windows%0A-%20OS%3A%20windows%0A-%20Architecture%3A%20x86_64%0A-%20Pointer%20width%3A%2064%0A-%20Endian%3A%20little%0A-%20CPU%20features%3A%20cmpxchg16b%2Cfxsr%2Csse%2Csse2%2Csse3%0A-%20Host%3A%20x86_64-pc-windows-msvc%0A%0A&build-config=%5Balias%5D%0Abr%20%3D%20%5B%22build%22%2C%20%22--release%22%5D%0Abrl%20%3D%20%5B%22build%22%2C%20%22--profile%22%2C%20%22release-lto%22%5D%0Arr%20%3D%20%5B%22run%22%2C%20%22--release%22%5D%0Arrl%20%3D%20%5B%22run%22%2C%20%22--profile%22%2C%20%22release-lto%22%5D%0Atr%20%3D%20%5B%22test%22%2C%20%22--release%22%5D%0Atrl%20%3D%20%5B%22test%22%2C%20%22--profile%22%2C%20%22release-lto%22%5D%0A%0A%5Bbuild%5D%0Arustflags%20%3D%20%5B%22-C%22%2C%20%22target-cpu%3Dnative%22%5D%0A

### Build Configuration

In `C:\Users\USERNAME\.cargo\config.toml` I have:

    [alias]
    br = "build --release"
    rr = "run --release"
    tr = "test --release"
    brl = "build --profile release-lto"
    rrl = "run --profile release-lto"
    trl = "test --profile release-lto"
    
    [build]
    rustflags = ["-C", "target-cpu=native"]

### Additional Context

I understand that this might already be a known problem, but hopefully this is a fairly small and simple real-world example that might be useful to have.

If I have misunderstood something, and it's correct that the lints trigger here I would of course love to know as well! :)

**Comment by obi1kenobi** [comment]
Thanks! You're correct that the lints shouldn't trigger here. We're using
https://github.com/obi1kenobi/cargo-semver-checks/issues/638 to track this,
and solving that issue will resolve this case as well. The analysis
currently just can't see items from a different crate, nor different
versions of the same crate.
